packages/transaction-controller/src/TransactionController.ts

```diff
@@ -175,6 +182,8 @@ export class TransactionController extends BaseController<
 
   private readonly isHistoryDisabled: boolean;
 
+  private readonly isSwapsDisabled: boolean;
```

Comment:
I thought processing a swaps transaction is a feature and disable/enable it explicitly, makes one easy to determine if this feature is available. Otherwise you need to go over all `addTransaction` in whole client code and maybe more in future.

Reply:
I agree it's a feature that we'd want to disable, but does the `swap` and `swapApproval` transaction type argument achieve the same thing as the client has to explicitly specify it's a swap transaction type when calling `addTransaction` for any of our new logic to be executed.

And mobile doesn't provide `transactionType` at all since we just added it to core.

At this point, it's a small gain so let's stick with the boolean for now.